src/MultiRegion/multi_region_cubed_sphere_grid.jl

```diff
@@ -204,183 +204,184 @@ function ConformalCubedSphereGrid(arch::AbstractArchitecture=CPU(), FT=Float64;
                                         η = region_η,
                                         rotation = region_rotation)
 
-    grid = MultiRegionGrid{FT, region_topology[1], region_topology[2], region_topology[3]}(arch,
-                                                                                           partition,
-                                                                                           connectivity,
-                                                                                           region_grids,
-                                                                                           devices)
+    grid = MultiRegionGrid{FT, region_topology...}(arch,
+                                                   partition,
+                                                   connectivity,
+                                                   region_grids,
+                                                   devices)
+
+    fields = (:λᶜᶜᵃ,   :φᶜᶜᵃ,   :Azᶜᶜᵃ )
+    LXs    = (:Center, :Center, :Center)
+    LYs    = (:Center, :Center, :Center)
+
+    for (field, LX, LY) in zip(fields, LXs, LYs)
+        expr = quote
+            $(Symbol(field)) = Field{$(Symbol(LX)), $(Symbol(LY)), Nothing}($(grid))
+
+            CUDA.@allowscalar begin
+                for region in 1:number_of_regions($(grid))
+                    getregion($(Symbol(field)), region).data .= getregion($(grid), region).$(Symbol(field))
+                end
+            end
+
+            if $(horizontal_topology) == FullyConnected
+                fill_cubed_sphere_halo_regions!($(Symbol(field)))
+            end
+
+            CUDA.@allowscalar begin
+                for region in 1:number_of_regions($(grid))
+                    getregion($(grid), region).$(Symbol(field)) .= getregion($(Symbol(field)), region).data
+                end
+            end
+        end # quote
+
+        eval(expr)
+    end
 
-    # TODO: Use fill_halo_regions! for the Face-Face-Any fields.
-    #       Need to add the Face-Face-Any coords/metric fiels in the tuples below and then abolish
-    #       the use of fill_faceface_coordinates!(grid) and fill_faceface_metrics!(grid)
+    field_1 = :Δxᶜᶜᵃ
+    LX_1    = :Center
+    LY_1    = :Center
+
+    field_2 = :Δyᶜᶜᵃ
+    LX_2    = :Center
+    LY_2    = :Center
```

Comment:
why are these done in a different loop  and not together with 

https://github.com/CliMA/Oceananigans.jl/blob/3da2e1613173149608a1acfa6e3bf312d61ebf8e/src/MultiRegion/multi_region_cubed_sphere_grid.jl#L274-L282

???